### PR TITLE
[WebcamAPI] Add Webcam API Docs

### PR DESCRIPTION
**Pull request by ly015**
## Motivation

- Refactor some interfaces of Webcam API towards user-friendliness
    - Move effect functions out from mmpose/core
    - Move pose index functions to tools/webcam/webcam_api/utils/pose.py
    - SingleFrameEffectNode -> FrameDrawingNode
- Add README and tutorials for Webcam API

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit tests to ensure correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: tools/webcam/docs/get_started_cn.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        ...]  # 更多节点
+)
+```
+
+### 核心概念：Runner 和 Node
+
+到这里，我们已经引出了 MMPose WebcamAPI 的２个最重要的概念：runner 和 Node，下面做正式介绍：
+
+- Runner：Runner 类是程序的主体，提供了程序启动的入口runner.run()方法，并负责视频读入，输出显示等功能。此外，runner 中会包含若干个 Node，分别负责在视频帧的处理中执行不同的功能。
+- Node：Node 类用来定义功能模块，例如模型推理，可视化，特效绘制等都可以通过定义一个对应的 Node 来实现。如上面的 config 例子中，2 个节点的功能分别是做目标检测（Detector）和姿态估计（TopDownPoseEstimator）
```
Comment (liqikai9) [inline tools/webcam/docs/get_started_cn.md:90]: TopDownPoseEstimator -> Human Pose Estimator

File: tools/webcam/docs/get_started_cn.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+一个重要的问题是：当一帧视频数据被 runner 读取后，会按照怎样的顺序通过所有的 Node 并最终被输出（显示）呢？
+答案就是 config 中每个 Node 的输入输出配置。如示例 config 中，可以看到每个 Node 都有`input_buffer`，`output_buffer`等参数，用来定义该节点的输入输出。通过这种连接关系，所有的 Node 构成了一个有向无环图结构，如下图所示：
+
+<div align=center>
+<img src="https://user-images.githubusercontent.com/15977946/153772900-9619ae80-3e64-4b40-bc1e-b184405e3d5b.png">
+</div>
+
+图中的每个 Data Buffer 就是一个用来存放数据的容器。用户不需要关注 buffer 的具体细节，只需要将其简单理解成 Node 输入输出的名字即可。用户在 config 中可以任意定义这些名字，不过要注意有以下几个特殊的名字：
+
+- _input_：存放 runner 读入的视频帧，用于模型推理
```
Comment (liqikai9) [inline tools/webcam/docs/get_started_cn.md:109]: _input_ -> \_input_,  _frame_ -> \_frame_, _display_ -> \_display_

File: tools/webcam/docs/get_started_cn.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+- _frame_ ：存放 runner 读入的视频帧，用于可视化
+- _display_：存放经过所以 Node 处理后的结果，用于在屏幕上显示
+
+当一帧视频数据被 runner 读入后，会被放进_input_和_frame_两个 buffer 中，然后按照 config 中定义的 Node 连接关系依次通过各个 Node ，最终到达_display_，并被 runner 读出显示在屏幕上。
+
+#### Get Advanced: 关于 buffer
+
+- Buffer 本质是是一个有限长度的队列，在 runner 中会包含一个 BufferManager 实例（见`mmpose/tools/webcam/webcam_api/buffer.py'）来生成和管理所有 buffer。Node 会按照 config 从对应的 buffer 中读出或写入数据。
```
Comment (liqikai9) [inline tools/webcam/docs/get_started_cn.md:117]: 是是 -> 是

Comment (liqikai9) [inline tools/webcam/docs/get_started_cn.md:117]: `mmpose/tools/webcam/webcam_api/buffer.py' -> [mmpose/tools/webcam/webcam_apis/utils/buffer.py](/tools/webcam/webcam_apis/utils/buffer.py)

File: tools/webcam/docs/example_cn.md [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+可以看到，这个 config 已经实现了我们设计思路中“1-目标检测”和“2-关键点检测”的功能。我们还需要实现“3-贴素材图”功能，这就需要定义一个新的 Node了。
+
+## Step 2：实现一个新 Node
+
+在 WebcamAPI 我们定义了以下 2 个 Node 基类：
+
+1. Node：所有 node 的基类，实现了初始化，绑定 runner，启动运行，数据输入输出等基本功能。子类通过重写抽象方法`process()`方法定义具体的 node 功能。
+2. FrameDrawingNode：用来绘制图像的 node 基类。FrameDrawingNode继承自 Node 并进一步封装了`process()`方法，提供了抽象方法`draw()`供子类实现具体的图像绘制功能。
+
+显然，“贴素材图”这个功能属于图像绘制，因此我们只需要继承 BaseFrameEffectNode 类即可。具体实现如下：
```
Comment (liqikai9) [inline tools/webcam/docs/example_cn.md:30]: BaseFrameEffectNode -> FrameDrawingNode

File: tools/webcam/docs/example_cn.md [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+2. 调用`process()`来执行具体的功能。`process()`是一个抽象接口，由子类具体实现
+    - 特别地，如果节点需要实现“开/关”功能，则还需要实现`bypass()`方法，以定义节点“关”时的行为。`bypass()`与`process()`的输入输出接口完全相同。在run()中会根据`Node.enable`的状态，调用`process()`或`bypass()`
+3. 将运行结果发送到输出 buffer
+
+在继承 Node 类实现具体的节点类时，通常需要完成以下工作：
+
+1. 在__init__()中注册输入、输出 buffer，并调用基类的__init__()方法
+2. 实现process()和bypass()（如需要）方法
+
+[FrameDrawingNode 类](tools/webcam/webcam_apis/nodes/frame_drawing_node.py) ：继承自 Node 类，对`process()`和`bypass()`方法做了进一步封装：
```
Comment (liqikai9) [inline tools/webcam/docs/example_cn.md:111]: tools/webcam/webcam_apis/nodes/frame_drawing_node.py -> /tools/webcam/webcam_apis/nodes/frame_drawing_node.py

File: tools/webcam/docs/example_cn.md [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+
+- process()：从接到输入中提取帧图像，传入draw()方法中绘图。draw()是一个抽象接口，有子类实现
+- bypass()：直接将节点输入返回
+
+### Get Advanced: 关于节点的输入、输出格式
+
+我们定义了[FrameMessage 类](tools/webcam/webcam_apis/utils/message.py)作为节点间通信的数据结构。也就是说，通常情况下节点的输入、输出和 buffer 中存储的元素，都是 FrameMessage 类的实例。FrameMessage 通常用来存储视频中1帧的信息，它提供了简单的接口，用来提取和存入数据：
```
Comment (liqikai9) [inline tools/webcam/docs/example_cn.md:118]: tools/webcam/webcam_apis/utils/message.py -> /tools/webcam/webcam_apis/utils/message.py

File: tools/webcam/README.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+* [Get started with MMPose Webcam API (Chinese)](/tools/webcam/docs/get_started_cn.md)
+* [Build a Webcam App: A Step-by-step Instruction (Chinese)](/tools/webcam/docs/example_cn.md)
+
+## Examples
+
+* [Pose Estimation](/tools/webcam/configs/examples/): A simple example to estimate and visualize human/animal pose.
+* [Eye Effects](/tools/webcam/configs/eyes/): Apply sunglasses and bug-eye effects.
+* [Face Swap](/tools/webcam/configs/face_swap/): Everybody gets someone else's face.
+* [Meow Dwen Dwen](/tools/webcam/configs/meow_dwen_dwen/): Dress up your cat in Bing Dwen Dwen custume.
```
Comment (liqikai9) [inline tools/webcam/README.md:25]: Does 'custume' mean 'costume' here?

File: tools/webcam/docs/get_started_cn.md (hunk 1 above)
Comment (ly015) [reply tools/webcam/docs/get_started_cn.md:92]: TopDownPoseEstimator is the name of a specific Node class.

File: tools/webcam/README.md (hunk 7 above)
Comment (ly015) [reply tools/webcam/README.md:26]: Yes, will fix.

**Pull request merged**